custom/icds_reports/utils/aggregation_helpers/distributed/delivery_forms.py

```diff
@@ -17,39 +17,47 @@ def aggregation_query(self):
         month = self.month.replace(day=1)
         current_month_start = month_formatter(self.month)
         next_month_start = month_formatter(self.month + relativedelta(months=1))
+        prev_month_start = month_formatter(self.month - relativedelta(months=1))
 
         query_params = {
             "month": month_formatter(month),
             "state_id": self.state_id,
             "current_month_start": current_month_start,
             "next_month_start": next_month_start,
+            "prev_month_start": prev_month_start
         }
 
         return """
         INSERT INTO "{tablename}" (
           case_id, state_id, supervisor_id, month, latest_time_end_processed,
-          breastfed_at_birth, valid_visits, where_born, num_children_del, still_live_birth
+          breastfed_at_birth, breastfed_at_birth_original_status, valid_visits, where_born, num_children_del, still_live_birth
         ) (
           SELECT
-            DISTINCT case_load_ccs_record0 AS case_id,
+            DISTINCT ucr.case_load_ccs_record0 AS case_id,
             %(state_id)s AS state_id,
-            LAST_VALUE(supervisor_id) over w as supervisor_id,
+            LAST_VALUE(ucr.supervisor_id) over w as supervisor_id,
             %(month)s::DATE AS month,
-            LAST_VALUE(timeend_with_time) over w AS latest_time_end_processed,
-            LAST_VALUE(breastfed_at_birth) over w as breastfed_at_birth,
-            SUM(CASE WHEN (unscheduled_visit=0 AND days_visit_late < 8) OR
-                          (timeend_with_time::DATE - next_visit) < 8 THEN 1 ELSE 0 END
+            LAST_VALUE(ucr.timeend_with_time) over w AS latest_time_end_processed,
+            LAST_VALUE(ucr.breastfed_at_birth) over w as breastfed_at_birth,
+            COALESCE(LAST_VALUE(ucr.breastfed_at_birth) over w , prev_month.breastfed_at_birth) as breastfed_at_birth_original_status,
+            SUM(CASE WHEN (ucr.unscheduled_visit=0 AND ucr.days_visit_late < 8) OR
+                          (ucr.timeend_with_time::DATE - ucr.next_visit) < 8 THEN 1 ELSE 0 END
                 ) OVER w as valid_visits,
-            LAST_VALUE(where_born) OVER w AS where_born,
-            LAST_VALUE(num_children_del) OVER w AS num_children_del,
-            LAST_VALUE(still_live_birth) OVER w AS still_live_birth
-          FROM "{ucr_tablename}"
-          WHERE state_id = %(state_id)s AND
-                timeend_with_time >= %(current_month_start)s AND timeend_with_time < %(next_month_start)s AND
-                case_load_ccs_record0 IS NOT NULL
+            LAST_VALUE(ucr.where_born) OVER w AS where_born,
+            LAST_VALUE(ucr.num_children_del) OVER w AS num_children_del,
+            LAST_VALUE(ucr.still_live_birth) OVER w AS still_live_birth
```

Comment:
would not it pick data from last months for these other columns as well as we are full joining with the previous month.